src/core/default_appearance.js

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isName, Name } from "./primitives.js";
+import { ColorSpace } from "./colorspace.js";
+import { escapePDFName } from "./core_utils.js";
+import { EvaluatorPreprocessor } from "./evaluator.js";
+import { OPS } from "../shared/util.js";
+import { StringStream } from "./stream.js";
+
+class DefaultAppearanceEvaluator extends EvaluatorPreprocessor {
+  constructor(str) {
+    super(new StringStream(str));
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: Name.get(""),
+      fontColor: new Uint8ClampedArray([0, 0, 0]) /* black */,
+    };
+
+    while (this.read(operation)) {
+      if (this.stateManager.stateStack.length !== 0) {
+        // Don't get info in save/restore sections.
+        continue;
+      }
+      const { fn, args } = operation;
+      switch (fn | 0) {
+        case OPS.setFont:
+          const [fontName, fontSize] = args;
+          if (isName(fontName)) {
+            result.fontName = fontName;
+          }
+          if (typeof fontSize === "number" && fontSize > 0) {
+            result.fontSize = fontSize;
+          }
+          break;
+        case OPS.setFillRGBColor:
+          ColorSpace.singletons.rgb.getRgbItem(args, 0, result.fontColor, 0);
+          break;
+        case OPS.setFillGray:
+          ColorSpace.singletons.gray.getRgbItem(args, 0, result.fontColor, 0);
+          break;
+        case OPS.setFillColorSpace:
+          ColorSpace.singletons.cmyk.getRgbItem(args, 0, result.fontColor, 0);
+          break;
+      }
+      args.length = 0;
+    }
```

Comment:
Given that the `Parser`/`Lexer` will throw on invalid data, which would cause Annotation-parsing to fail, please wrap all of this code in a try-catch block as below:

```suggestion
    try {
      while (this.read(operation)) {
        if (this.stateManager.stateStack.length !== 0) {
          // Don't get info in save/restore sections.
          continue;
        }
        const { fn, args } = operation;
        switch (fn | 0) {
          case OPS.setFont:
            const [fontName, fontSize] = args;
            if (isName(fontName)) {
              result.fontName = fontName;
            }
            if (typeof fontSize === "number" && fontSize > 0) {
              result.fontSize = fontSize;
            }
            break;
          case OPS.setFillRGBColor:
            ColorSpace.singletons.rgb.getRgbItem(args, 0, result.fontColor, 0);
            break;
          case OPS.setFillGray:
            ColorSpace.singletons.gray.getRgbItem(args, 0, result.fontColor, 0);
            break;
          case OPS.setFillColorSpace:
            ColorSpace.singletons.cmyk.getRgbItem(args, 0, result.fontColor, 0);
            break;
        }
        args.length = 0;
      }
    } catch (reason) {
      warn(`parseDefaultAppearance - ignoring errors: "${reason}".`);
    }
```